Fix readonly check in libmdbx-rs

wtdcode opened the pull request:
The previous RO/RW check of `libmdbx-rs` is obviously wrong because read-write mode doesn't imply `WriteMap`. For example, it is perfectly valid to open an environment without `WriteMap` and in this case `is_read_write` will return false.

```rust
    /// Returns true if the environment was opened in [`crate::Mode::ReadWrite`] mode.
    #[inline]
    pub fn is_read_write(&self) -> bool {
        self.inner.env_kind.is_write_map()
    }

    /// Returns true if the environment was opened in [`crate::Mode::ReadOnly`] mode.
    #[inline]
    pub fn is_read_only(&self) -> bool {
        !self.inner.env_kind.is_write_map()
    }
```

Since reth will always open the database with `WriteMap`, `is_read_only` will always return `false`, which _happens_ to pass the check in `crates/cli/commands/src/common.rs`. `is_read_write` is not referenced anywhere and this error on `is_read_write` doesn't have any effect.

This pull request fixes the issue by asking the actual mode from MDBX and adds a few tests to reflect this change.

shekhirin reviewed (approved):
The change and tests make sense and LGTM, thank you! There's a bunch of semicolons that sneaked into the PR, can you please undo them?

wtdcode commented:
Fixed clippy and undo the unrelated changes. I hope CI is happy.

I also fixed `mdbx_txn_renew` import by the way because it should be only imported with `read-tx-timeouts` feature. This broke my `cargo +nightly clippy --fix` just now.

wtdcode commented:
Also fixed docs accordingly.
